src/strings/ops.c

```diff
@@ -544,9 +544,17 @@ static MVMint64 MVM_string_memmem_grapheme32 (MVMThreadContext *tc, MVMGrapheme3
 static MVMint64 MVM_string_memmem_grapheme32str (MVMThreadContext *tc, MVMString *Haystack, MVMString *needle, MVMint64 H_start, MVMStringIndex H_graphs, MVMStringIndex n_graphs) {
     MVMGrapheme32 *needle_buf = NULL;
     MVMint64 rtrn;
+    int needle_is_malloced = 0;
     if (needle->body.storage_type != MVM_STRING_GRAPHEME_32) {
         MVMStringIndex i;
-        needle_buf = MVM_malloc(needle->body.num_graphs * sizeof(MVMGrapheme32));
+        size_t needle_size = (1 + n_graphs) * sizeof(MVMGrapheme32);
+        /* Allocate max 8K onto the stack, otherwise malloc */
+        if (needle_size < 3000)
```

Comment:
Comment says 8K, allocation amount is 3K?

Reply:
That was just copied from `knuth_morris_pratt_string_index()`, looks like the comment is wrong there too. I'll correct the comments and force push.